### PR TITLE
support both octavia and old lbaasv2 api in openstack

### LINKED ISSUE #6437: support openstack octavia
**1. Describe IN DETAIL the feature/behavior/change you would like to see.**

I would like to see support to openstack lbaas octavia

**2. Feel free to provide a design supporting your feature request.**

Instead of using hardcoded loadbalancer endpoint, we should do it like this https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/cloudprovider/providers/openstack/openstack_client.go#L81-L98

### PR DESCRIPTION
**Pull request by zetaab**
fixes #6437 

/sig openstack

### CODE REVIEW

File: upup/pkg/fi/cloudup/openstack/cloud.go [hunk 1]
```diff
@@ -388,8 +380,30 @@ func NewOpenstackCloud(tags map[string]string, spec *kops.ClusterSpec) (Openstac
...
+	if octavia {
+		glog.V(2).Infof("Openstack using Octavia loadbalancer")
+		lbClient, err = os.NewLoadBalancerV2(provider, gophercloud.EndpointOpts{
+			Region: region,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("error building lb client: %v", err)
+		}
+	} else {
+		glog.V(2).Infof("Openstack using old lbaasv2 loadbalancer")
```
Comment (chrisz100) [inline upup/pkg/fi/cloudup/openstack/cloud.go]: nit: I think mentioning the other loadbalancer as "old" is a little harsh. Its either an alternative or deprecated.

**Comment by zetaab** [comment]
/hold

There is one bug that we should always wait that LB is in active state before we can add listener/pools to it. Currently it will return lots of errors because of this. I will fix this before this is merged

**Comment by zetaab** [comment]
/hold cancel

now its ready and looks like following:

```
I0205 16:29:19.843784   52100 executor.go:103] Tasks: 71 done / 79 total; 4 can run
I0205 16:29:20.558386   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:29:21.610390   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:29:22.856263   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:29:24.337667   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:29:25.375800   52100 context.go:231] hit maximum retries 4 with error GetFloatingIP: fetching floating IP failed: Resource not found
I0205 16:29:26.101690   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:29:28.230764   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:29:30.760928   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:29:33.799347   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:29:37.421187   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:29:41.761765   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:29:46.975305   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:29:53.210504   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:30:00.697665   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:30:09.644859   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:30:20.386935   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:30:33.255127   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:30:48.707058   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
I0205 16:31:07.249153   52100 lb.go:74] Waiting for Loadbalancer to be ACTIVE...
W0205 16:31:29.590818   52100 executor.go:130] error running task "Keypair/master" (7m50s remaining to succeed): error finding address for *openstacktasks.FloatingIP {"Name":"fip-api.oma.k8s.local","ID":"0004ffc1-3acb-4080-bb46-603f82166cca","Server":null,"LB":{"ID":"a9a56dd7-233e-4e3e-9cbf-66078058d8d9","Name":"api.oma.k8s.local","Subnet":"nova.oma.k8s.local","VipSubnet":"8cc71087-fee9-4975-a05e-96b2df335fc8","Lifecycle":"Sync","PortID":"eade07b7-7291-48b0-8322-dde0c0596aca"},"Lifecycle":"Sync"}: GetFloatingIP: fetching floating IP failed: Resource not found
I0205 16:31:29.590895   52100 executor.go:103] Tasks: 74 done / 79 total; 5 can run
I0205 16:31:30.262119   52100 vfs_castore.go:729] Issuing new certificate: "master"
I0205 16:31:36.646586   52100 executor.go:103] Tasks: 79 done / 79 total; 0 can run
I0205 16:31:36.899274   52100 update_cluster.go:290] Exporting kubecfg for cluster
kops has set your kubectl context to oma.k8s.local

Cluster is starting.  It should be ready in a few minutes.
```

(some errors there but those are fixed in another PRs). So it will print message when its waiting that LB is ready. It takes sometimes (about 2 minutes for me in devstack)

**Comment by mikesplain** [comment]
Thanks @zetaab!

/lgtm
/approve

**Pull request merged**